client/sections-middleware.js

```diff
@@ -0,0 +1,103 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import config from 'config';
+import page from 'page';
+import { find, filter, isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { activateNextLayoutFocus } from 'state/ui/layout-focus/actions';
+import * as LoadingError from 'layout/error';
+import * as controller from './controller/index.web';
+import { restoreLastSession } from 'lib/restore-last-path';
+import { hub as preloadHub } from 'sections-preload';
+import { switchCSS } from 'lib/i18n-utils/switch-locale';
+import { pathToRegExp } from './utils';
+
+import sections from './sections';
+
+const _loadedSections = {};
+
+function maybeLoadCSS( sectionName ) {
+	//eslint-disable-line no-unused-vars
+	const section = find( sections, { name: sectionName } );
+
+	if ( ! ( section && section.css ) ) {
+		return;
+	}
+
+	const url =
+		typeof document !== 'undefined' && document.documentElement.dir === 'rtl'
+			? section.css.urls.rtl
+			: section.css.urls.ltr;
+
+	switchCSS( 'section-css-' + section.css.id, url );
+}
+
+function preload( sectionName ) {
+	maybeLoadCSS( sectionName );
+	const filteredSections = filter( sections, { name: sectionName } );
+	if ( isEmpty( sections ) ) {
+		return Promise.reject( `Attempting to load non-existent section: ${ sectionName }` );
+	}
+	return Promise.all( filteredSections.map( section => section.load() ) );
+}
+
+preloadHub.on( 'preload', preload );
+
+function activateSection( sectionDefinition, context, next ) {
+	const dispatch = context.store.dispatch;
```

Comment:
Nitpick -- Can we destructure here? Or is that part of the language spec not available at this stage?

Reply:
🤷‍♂️

Reply:
If the question is "Does Caylpso's Node version (8.9.3 at the moment) support destructuring assignment?" the answer is "Yes."